CodeGeneration/Sources/generate-swiftsyntaxbuilder/BuildableNodesFile.swift

```diff
@@ -76,12 +92,73 @@ let buildableNodesFile = SourceFile {
         createWithTrailingCommaFunction()
       }
       for trivia in trivias {
-        createWithTriviaFunction(trivia: trivia)
+        createWithTriviaFunction2(trivia: trivia)
       }
     }
   }
 }
 
+private func createHasTrailingCommaVariable() -> VariableDecl {
+  VariableDecl(name: "hasTrailingComma", type: "Bool") {
+    SwitchStmt(expression: "data", cases: SwitchCaseList([
+      SwitchCase(
+        label: SwitchCaseLabel(
+          caseItems: CaseItem(
+            pattern: ExpressionPattern(
+              expression: FunctionCallExpr(MemberAccessExpr(name: "buildable")) {
+                TupleExprElement(
+                  expression: UnresolvedPatternExpr(
+                    pattern: ValueBindingPattern(
+                      letOrVarKeyword: .let,
+                      valuePattern: IdentifierPattern("buildableData")
+                    )
+                  )
+                )
+              }
+            )
+          )
+        ),
+        statements: CodeBlockItemList {
+          CodeBlockItem(
+            item: ReturnStmt(expression: SequenceExpr {
+              MemberAccessExpr(base: "buildableData", name: "trailingComma")
+              BinaryOperatorExpr("!=")
+              NilLiteralExpr()
+            })
+          )
+        }
+      ),
+      SwitchCase(
+        label: SwitchCaseLabel(
+          caseItems: CaseItem(
+            pattern: ExpressionPattern(
+              expression: FunctionCallExpr(MemberAccessExpr(name: "constructed")) {
+                TupleExprElement(
+                  expression: UnresolvedPatternExpr(
+                    pattern: ValueBindingPattern(
+                      letOrVarKeyword: .let,
+                      valuePattern: IdentifierPattern("node")
+                    )
+                  )
+                )
+              }
+            )
+          )
+        ),
+        statements: CodeBlockItemList {
+          CodeBlockItem(
+            item: ReturnStmt(expression: SequenceExpr {
+              MemberAccessExpr(base: "node", name: "trailingComma")
+              BinaryOperatorExpr("!=")
+              NilLiteralExpr()
+            })
+          )
+        }
+      )
+    ]))
```

Comment:
These switch statements can be written a bit more concisely using the builder-based initializers. I usually also keep the `label/caseItems/pattern/expression` boilerplate on a single line, but perhaps there is some other way too to avoid the pyramid of doom here.

```suggestion
    SwitchStmt(expression: "data") {
      SwitchCase(
        label: SwitchCaseLabel(
          caseItems: CaseItem(
            pattern: ExpressionPattern(
              expression: FunctionCallExpr(MemberAccessExpr(name: "buildable")) {
                TupleExprElement(
                  expression: UnresolvedPatternExpr(
                    pattern: ValueBindingPattern(
                      letOrVarKeyword: .let,
                      valuePattern: IdentifierPattern("buildableData")
                    )
                  )
                )
              }
            )
          )
        )
      ) {
        ReturnStmt(expression: SequenceExpr {
          MemberAccessExpr(base: "buildableData", name: "trailingComma")
          BinaryOperatorExpr("!=")
          NilLiteralExpr()
        })
      }
      SwitchCase(
        label: SwitchCaseLabel(
          caseItems: CaseItem(
            pattern: ExpressionPattern(
              expression: FunctionCallExpr(MemberAccessExpr(name: "constructed")) {
                TupleExprElement(
                  expression: UnresolvedPatternExpr(
                    pattern: ValueBindingPattern(
                      letOrVarKeyword: .let,
                      valuePattern: IdentifierPattern("node")
                    )
                  )
                )
              }
            )
          )
        )
      ) {
        ReturnStmt(expression: SequenceExpr {
          MemberAccessExpr(base: "node", name: "trailingComma")
          BinaryOperatorExpr("!=")
          NilLiteralExpr()
        })
      }
    }
  }
```

Reply:
Thanks. That’s a little cleaner. I’ll probably migrate all of this to be parser-based in a follow-up PR.